Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -284,16 +287,147 @@ class NexusGeometrySaveTest : public CxxTest::TestSuite {
     saveInstrument(compInfo, destinationFile); // saves instrument
     HDF5FileTestUtility testUtility(destinationFile);
 
-    // expectedInstrumentName; // let dataset name have same value as component
-    // name.
+    std::string expectedDataSetValue = "test_data_for_instrument";
 
     TS_ASSERT(testUtility.hasDataSet(expectedInstrumentName,
+                                     expectedDataSetValue,
                                      "/raw_data_1/instrument"));
 
     TS_ASSERT(testUtility.hasAttributeInDataSet(
         expectedInstrumentName, "/raw_data_1/instrument", SHORT_NAME,
         expectedInstrumentName));
   }
+
+  void test_translation() {
+
+
+    auto instrument =
+        ComponentCreationHelper::createTestInstrumentRectangular2(1, 10);
+
+    auto beamline =
+        Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+    auto &compInfo = (*beamline.first);
+
+    // translate source
+    Eigen::Vector3d absoluteBankLocation(5, 0, 0);
+
+    compInfo.setPosition(compInfo.indexOfAny("bank1"),
+                         Mantid::Kernel::toV3D(absoluteBankLocation));
+
+    auto relativeSourcePosition = Mantid::Kernel::toVector3d(
```

Comment:
The naming here doesn't make sense. You are getting the relative position of the bank but calling it the `relativeSourcePostition`?